### DISCUSSION TITLE
Only a single InvoiceDocumentReference in xml file

### DISCUSSION THREAD

**Discussion by digikungfu** [stephanstapel/ZUGFeRD-csharp#507] [General]
Hi there,
if a add multiple InvoiceDocumentReference s  with desc.AddInvoiceReferencedDocument() I can only see the first one in the generated xml document. My profile is  

desc.Save(
                stream: stream, 
                version: ZUGFeRDVersion.Version23,
                profile: Profile.XRechnung,
                format: ZUGFeRDFormats.UBL
                );
                
Here are the first lines of the Result:
<?xml version="1.0" encoding="utf-8"?>
<ubl:Invoice xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:ext="urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ubl="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2">
  <cbc:CustomizationID>urn:cen.eu:en16931:2017#compliant#urn:xeinkauf.de:kosit:xrechnung_3.0</cbc:CustomizationID>
  <cbc:ProfileID>urn:fdc:peppol.eu:2017:poacc:billing:01:1.0</cbc:ProfileID>
  <cbc:ID>A K24/552</cbc:ID>
  <cbc:IssueDate>2024-08-30</cbc:IssueDate>
  <cbc:DueDate>2024-09-17</cbc:DueDate>
  <cbc:InvoiceTypeCode>380</cbc:InvoiceTypeCode>
  <cbc:Note>Rechnung zu unserem Projekt: </cbc:Note>
  <cbc:DocumentCurrencyCode>EUR</cbc:DocumentCurrencyCode>
  <cac:BillingReference>
    <cac:InvoiceDocumentReference>
      <cbc:ID>A K21/119</cbc:ID>
      <cbc:IssueDate>2023-12-31</cbc:IssueDate>
    </cac:InvoiceDocumentReference>
  </cac:BillingReference>
  <cac:AccountingSupplierParty>
  
  Anny hints?
  
  regards
  Andy

**Comment by stephanstapel** [comment]
Hi @digikungfu ,

UBL supports only one such reference:

https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-BillingReference/cac-InvoiceDocumentReference/

**Reply by StefanF-Austria** [reply]
Hi @stephanstapel ,

> Hi @digikungfu ,
> 
> UBL supports only one such reference:
> 
> https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-BillingReference/cac-InvoiceDocumentReference/

There can be only one InvoiceDocumentReference, that's right.
BUT
There can be "n" BillingReferences
https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-BillingReference/
and the BillingReference only includes the property InvoiceDocumentReference

Stefan

**Comment by digikungfu** [comment]
Thanks for your reply,

I was wondering, because we try some commercial tools witch starts with

<?xml version="1.0" encoding="UTF-8"?>
<ns0:Invoice xmlns:ns0="urn:oasis:names:specification:**ubl**:schema:xsd:Invoice-2"

that contains multiple InvoiceDocumentReferences.

If I switch the save parameter from ZUGFeRDFormats.UBL to ZUGFeRDFormats.CII I got a System.NullReferenceException. I guess there is some mandantory field I missed

**Comment by digikungfu** [comment]
I was able to solve the System.NullReferenceExeption error from my last post. I had included the ZUGFeRD component via nuget as a reference in my project. For further debugging, the idea was to delete the reference and instead include the complete source code, i.e. all cs files in the project.  That was the solution. I can now use the parameter ZUGFeRDFormats.CII in the Save method without the System.NullReferenceExeption error. Why this didn't work with nuget and referencing is unclear to me. Now I can also see all InvoiceDocumentReferences.
